### PR TITLE
containers: minor improvements to allow_selected_insecure_registries

### PR DESCRIPTION
**Pull request by asmorodskyi**
1. Remove creation of /etc/docker folder because actually we would like to fail in such cases
2. Make more clear error message in case unexpected runtime selected
3. Use handy function instead of direct call for sed

### CODE REVIEW

**Comment by asmorodskyi** [comment]
@Vogtinator  can you advice me which jobs I should clone from o3 to make sure that I haven't broke anything there ?

**Comment by asmorodskyi** [comment]
containers_basic from osd - http://autobot.qa.suse.de/tests/2235

**Comment by Vogtinator** [comment]
> @Vogtinator can you advice me which jobs I should clone from o3 to make sure that I haven't broke anything there ?

15.2 jeos-containers: https://openqa.opensuse.org/tests/1662177
TW extra_tests_textmode_containers: https://openqa.opensuse.org/tests/1663034
MicroOS staging: https://openqa.opensuse.org/tests/1664002

That should cover most uses

**Review by pdostal — APPROVED** [review]
I like the `file_content_replace` 🚀 thank you Anton!

**Comment by asmorodskyi** [comment]
> > @Vogtinator can you advice me which jobs I should clone from o3 to make sure that I haven't broke anything there ?
> 
> 15.2 jeos-containers: https://openqa.opensuse.org/tests/1662177
> TW extra_tests_textmode_containers: https://openqa.opensuse.org/tests/1663034
> MicroOS staging: https://openqa.opensuse.org/tests/1664002
> 
> That should cover most uses

http://autobot.qa.suse.de/tests/2236 - for 15.2 jeos-containers
http://autobot.qa.suse.de/tests/2237 - for TW extra_tests_textmode_containers

**Comment by asmorodskyi** [comment]
> containers_basic from osd - http://autobot.qa.suse.de/tests/2235

this one failed in docker module with unrelated reason but I decided to re-run it anyway http://autobot.qa.suse.de/tests/2240

**Comment by asmorodskyi** [comment]
> > @Vogtinator can you advice me which jobs I should clone from o3 to make sure that I haven't broke anything there ?
> 
> 15.2 jeos-containers: https://openqa.opensuse.org/tests/1662177
> TW extra_tests_textmode_containers: https://openqa.opensuse.org/tests/1663034
> MicroOS staging: https://openqa.opensuse.org/tests/1664002
> 
> That should cover most uses

http://autobot.qa.suse.de/tests/2241 - for MicroOS staging

**Review by b10n1k — APPROVED** [review]
+1

File: lib/containers/common.pm [hunk 1]
```diff
@@ -112,18 +112,16 @@ sub allow_selected_insecure_registries {
...
         assert_script_run('cat /etc/docker/daemon.json');
         systemctl('restart docker');
     } elsif ($runtime =~ /podman/) {
         assert_script_run "curl " . data_url('containers/registries.conf') . " -o /etc/containers/registries.conf";
         assert_script_run "chmod 644 /etc/containers/registries.conf";
-        assert_script_run("sed -i 's/REGISTRY/$registry/' /etc/containers/registries.conf");
+        file_content_replace("/etc/containers/registries.conf", REGISTRY => $registry);
     } else {
-        die "You must define the runtime!";
+        die "Not supported runtime - " . $runtime;
```
Comment (Vogtinator) [inline lib/containers/common.pm:124]:
```suggestion
        die "Unsupported runtime - " . $runtime;
```

**Review by Vogtinator — CHANGES_REQUESTED** [review]
Nitpick, looks good otherwise

File: lib/containers/common.pm (hunk 1 above)
Comment (asmorodskyi) [reply lib/containers/common.pm]: done

**Review by Vogtinator — APPROVED** [review]

**Pull request merged**